.travis.yml

```diff
@@ -43,11 +43,15 @@ cache:
     - $HOME/.npm
 
 install:
-  - travis_retry composer self-update && composer --version
-  - export PATH="$HOME/.composer/vendor/bin:$PATH"
-  - travis_retry composer install --prefer-dist --no-interaction
+  # setup PHP tests
+  - |
+    if [ $TASK_TESTS_PHP == 1 ]; then
```

Comment:
I think we need that for JS tests as well...

Reply:
i.e. JS packages are installed via composer.

Reply:
Good point :) The question is how it will be handled in 2.1 ...

As laid out in https://github.com/yiisoft/yii2/issues/14862#issue-259192470 I think we should drop requirements in `composer.json` in favour of `packages.json`.

Since I've added it to `packages.json`, the `npm install` command fetches the packages needed for testing.

Reply:
@klimov-paul any ideas about it?

Reply:
How should I know?
It seems you and @schmunk42 have already decided to move towards new asset plugin. If you did, you should start to upgrade infrastructure, including the docs at the core and all asset-related extensions.

I would rather use CDN as default asset bundle source to aviod necessity of any plugin or other composer hack usage. This will allow particular extension to function out-of-the-box in most case, while leaving the developer choice in which way he want to install the assets.

I suppose it is too early to perform global asset management changes, like enforcing 'foxy' usage.
The core extensions `yii2-gii` and `yii2-debug` should be stripped from asset dependency first. Only after then this matter can be considered.

Reply:
I answered here: https://github.com/yiisoft/yii2/issues/14862#issuecomment-369218596